hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -482,18 +481,19 @@ protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
       throw new RuntimeException("Interrupted during commit", e);
     }
 
-    // Otherwise stick to the {WAREHOUSE_DIR}/{DB_NAME}.db/{TABLE_NAME} path
-    String warehouseLocation = getWarehouseLocation();
-    return String.format(
-        "%s/%s.db/%s",
-        warehouseLocation, tableIdentifier.namespace().levels()[0], tableIdentifier.name());
+    // Otherwise, stick to the {WAREHOUSE_DIR}/{DB_NAME}.db/{TABLE_NAME} path
+    String warehouseLocation = getWarehouseLocation(tableIdentifier.namespace().levels()[0]);
+    return String.format("%s/%s", warehouseLocation, tableIdentifier.name());
   }
 
-  private String getWarehouseLocation() {
+  private String getWarehouseLocation(String warehouseName) {
     String warehouseLocation = conf.get(HiveConf.ConfVars.METASTOREWAREHOUSE.varname);
     Preconditions.checkNotNull(
         warehouseLocation, "Warehouse location is not set: hive.metastore.warehouse.dir=null");
-    return warehouseLocation;
+    if (warehouseLocation.endsWith("/")) {
+      warehouseLocation = warehouseLocation.substring(0, warehouseLocation.length() - 1);
+    }
```

Comment:
[minor] can use [LocationUtil#stripTrailingSlash](https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/util/LocationUtil.java#L26) .

Reply:
Nice, wasn't aware of that one. Updated it right away, thanks!